tools/gfx/flag-combiner.h

```diff
@@ -0,0 +1,62 @@
+#ifndef GFX_FLAG_COMBINER_H
+#define GFX_FLAG_COMBINER_H
+
+#include "../../source/core/list.h"
+
+namespace gfx {
+
+/* A default set of flags that can be used for checking devices */
+typedef uint32_t DeviceCheckFlags;
+struct DeviceCheckFlag
+{
+    enum Enum : DeviceCheckFlags
+    {
+        UseFullFeatureLevel = 0x1,      //< If set will use full feature level (on dx this is  D3D_FEATURE_LEVEL_11_1 else will try D3D_FEATURE_LEVEL_11_0)
+        UseHardwareDevice = 0x2,        //< If set will try a hardware device
+        UseDebug = 0x4,                 //< If set will enable use of debug
```

Comment:
Do we ever have cases where `UseDebug` really needs to go through the `FlagCombiner`? It seems like we always either request a debug device, or not, and we don't want to fall back from one to the other, right? Or are there cases where initializing the debug runtime could fail, so we *have* to fall back?

Reply:
Yes - and that was what drove the change. On the CI system system that things were failing on, it would fail because it did not have the debug capability.